packages/shell-api/src/shell-internal-state.ts

```diff
@@ -0,0 +1,238 @@
+    Object.assign(contextObject, ShellBson);
+    contextObject.rs = new ReplicaSet(this.currentDb.mongo);
+    contextObject.sh = new Shard(this.currentDb.mongo);
+
+    // Add global shell objects
+    this.asyncWriter.symbols.initializeApiObjects({
+      db: signatures.Database,
+      rs: signatures.ReplicaSet,
+      sh: signatures.Shard,
+      Mongo: signatures.Mongo
+    });
+
+    try {
+      Object.defineProperty(contextObject, 'db', {
+        set: (newDb: any) => {
+          if (newDb.shellApiType && newDb.shellApiType() !== 'Database') {
+            throw new MongoshInvalidInputError('Cannot reassign \'db\' to non-Database type');
+          }
+          this.currentDb = newDb;
+          contextObject.rs = new ReplicaSet(this.currentDb.mongo);
+          contextObject.sh = new Shard(this.currentDb.mongo);
+          this.fetchConnectionInfo();
+          return newDb;
+        },
+        get: () => {
+          return this.currentDb;
+        }
+      });
+    } catch (e) { // java shell, can't use getters/setters
+      console.log(`in java shell, platform: ${this.initialServiceProvider.platform}`);
+      contextObject.db = this.currentDb;
+    }
+
+    contextObject.Mongo = async(uri?, options?): Promise<Mongo> => {
+      if (
+        this.initialServiceProvider.platform !== ReplPlatform.CLI
+      ) {
+        throw new MongoshUnimplementedError(
+          `new Mongo connection are not supported for current platform: ${ReplPlatform[this.initialServiceProvider.platform]}`
+        );
+      }
+      const mongo = new Mongo(this, uri, options);
+      await mongo.connect();
+      return mongo;
+    };
+
+    this.messageBus.emit(
+      'mongosh:setCtx',
+      { method: 'setCtx', arguments: { db: contextObject.db } }
+    );
+  }
+
+  public emitApiCall(event: {
+    method: string;
+    class: string;
+    arguments: Document;
+    [otherProps: string]: any;
+  }): void {
+    this.messageBus.emit('mongosh:api-call', event);
+  }
+
+  /**
+   * Helper method to adapt aggregation pipeline options.
+   * This is here so that it's not visible to the user.
+   *
+   * @param options
+   */
+  public adaptAggregateOptions(options: any = {}): {
+    providerOptions: Document;
+    dbOptions: DatabaseOptions;
+    explain: boolean;
+  } {
+    const providerOptions = { ...options };
+
+    const dbOptions: DatabaseOptions = {};
+    let explain = false;
+
+    if ('readConcern' in providerOptions) {
+      dbOptions.readConcern = options.readConcern;
+      delete providerOptions.readConcern;
+    }
+
+    if ('writeConcern' in providerOptions) {
+      Object.assign(dbOptions, options.writeConcern);
+      delete providerOptions.writeConcern;
+    }
+
+    if ('explain' in providerOptions) {
+      explain = providerOptions.explain;
+      delete providerOptions.explain;
+    }
+
+    return { providerOptions, dbOptions, explain };
+  }
+
+  public validateExplainableVerbosity(verbosity: string): void {
```

Comment:
same for this one, i think it does not need to be dynamically passed down, it can be imported from a module where it is used.